python/ray/autoscaler/_private/autoscaler.py

```diff
@@ -349,15 +344,31 @@ def reset(self, errors_fatal=False):
             if not self.provider:
                 self.provider = _get_node_provider(self.config["provider"],
                                                    self.config["cluster_name"])
-            # Check whether we can enable the resource demand scheduler.
-            if "available_node_types" in self.config:
-                self.available_node_types = self.config["available_node_types"]
-                self.resource_demand_scheduler = ResourceDemandScheduler(
+
+            self.available_node_types = self.config["available_node_types"]
+            upscaling_speed = self.config.get("upscaling_speed")
+            aggressive = self.config.get("autoscaling_mode") == "aggressive"
+            target_utilization_fraction = self.config.get(
+                "target_utilization_fraction")
+            if upscaling_speed:
+                upscaling_speed = float(upscaling_speed)
+            elif aggressive:
+                upscaling_speed = float(self.config["max_workers"])
```

Comment:
This is somewhat different from the original notion of aggressive. Can you add a TODO to maybe support this in the future?

We would need an option like ``initial_upscaling_num_workers`` (currently hardcoded to 5). We can decide whether this is needed based on user feedback.

Reply:
Just wanted to clarify that setting upscaling_speed to max_workers is equivalent to setting it to infinity. Does it make sense?

Reply:
I am actually confused by what you mean by "original notion of aggressiveness". In the legacy autoscaler it just means that in the aggressive mode we upscale to initial_workers in case of load (not max regardless of load).

What you propose as an option seems orthogonal to scaling to max regardless of load. I added a todo now for `initial_upscaling_num_workers`.

Reply:
Btw if you set upscaling_speed to initial_upscaling_num_workers, at the first scale-up (when only the head node is there) you actually get initial_upscaling_num_workers. But I guess someone might want them to have different values. I added a TODO for now.

Reply:
Now that I think about it, why do we actually need `initial_upscaling_num_workers`? What new behavior it will introduce that is not currently covered in `upscaling_speed`? at the end of the day they both are constrained by demand and at the end the cluster will scale to the same size regardless of `upscaling_speed` and `initial_upscaling_num_workers`.
It feels redundant to me.

Reply:
Aggressive used to immediately scale to max regardless of demand. We don't do that any more since we always exactly  match demand.

Reply:
Re: second question, cluster size will be y = (1 + b)*speed^t; initial upscaling constant affects how long it takes to get to that size. Could be annoying to wait for slow autoscaling if you have huge clusters.